app/services/cold_storage_backends/s3_backend.py

```
@@ -0,0 +1,155 @@
+    def backend_name(self) -> str:
+        return "s3"
+
+    def capabilities(self) -> ColdStorageCapabilities:
+        return ColdStorageCapabilities(
+            supports_move=True,
+            supports_copy=True,
+            supports_symlink=False,
+            supports_local_path_stats=False,
+        )
+
+    def _client(self, location: ColdStorageLocation):
```

Ruff (0.15.10)

[warning] Missing return type annotation for private function `_client`

(ANN202)